pytorch_lightning/trainer/trainer.py

```diff
@@ -132,7 +132,7 @@ def __init__(self,
         self.gradient_clip_val = gradient_clip_val
         self.check_val_every_n_epoch = check_val_every_n_epoch
         self.track_grad_norm = track_grad_norm
-        self.on_gpu = gpus is not None and torch.cuda.is_available()
+        self.on_gpu = True if (gpus and torch.cuda.is_available()) else False
```

Comment:
you can just write `self.on_gpu = (gpus and torch.cuda.is_available())` as it is boolean statement

Reply:
I'm not  sure about that :
`print(([] and True))` --> `[]`
`print((0 and True))` --> `0`
Am I mistaken?

Reply:
you are right, I have overlooked that `gpus` is list